test/validation/api/scheduler/scheduler.c

```diff
@@ -2351,6 +2432,32 @@ static void scheduler_test_pause_enqueue(void)
 	CU_ASSERT(odp_queue_destroy(queue) == 0);
 }
 
+static void scheduler_test_private_groups(void)
+{
+	odp_shm_t shm;
+	thread_args_t *args;
+	test_globals_t *globals;
+	void *arg_ptr;
+
+	shm = odp_shm_lookup(GLOBALS_SHM_NAME);
+	CU_ASSERT_FATAL(shm != ODP_SHM_INVALID);
+	globals = odp_shm_addr(shm);
+	CU_ASSERT_FATAL(globals != NULL);
+
+	shm = odp_shm_lookup(SHM_THR_ARGS_NAME);
+	CU_ASSERT_FATAL(shm != ODP_SHM_INVALID);
+	args = odp_shm_addr(shm);
+	CU_ASSERT_FATAL(args != NULL);
+
+	args->globals = globals;
+
+	/* One worker per CPU creates its own group and validates scheduling */
+	arg_ptr = args;
+	CU_ASSERT_FATAL(odp_cunit_thread_create(globals->num_workers, private_group_thread,
```

Comment:
The number of supported schedule groups (`odp_schedule_capability_t.max_groups`) may be smaller than the number of available workers. Note that this may effect the barrier inside the test.